presenter/GraphSynthesizer.py

```diff
@@ -0,0 +1,8 @@
+def get_book_identifier(row) -> str:
+    if "book_name" in row:
+        return row["book_name"]
+    return row["book_id"]
```

Comment:
Again this seems very unintuitive, try to do better than that and consider if you even need a function.

Reply:
I have decided to inline it to the parent, possibly into a 1 liner, I have to see how readable it is in order to decide wether it should be a 1 liner or not.